runtime/moonbase/src/lib.rs

```diff
@@ -1102,7 +1102,7 @@ impl pallet_maintenance_mode::Config for Runtime {
 	type MaintenanceDmpHandler = MaintenanceDmpHandler;
 	// We use AllPalletsReversedWithSystemFirst because we dont want to change the hooks in normal
 	// operation
-	type NormalExecutiveHooks = AllPalletsReversedWithSystemFirst;
+	type NormalExecutiveHooks = AllPalletsWithSystem;
```

Comment:
I'm not sure what the implications of this are. The deprecation warning this change came from isn't very clear about why it should be avoided:

```rust
#[deprecated(note = "Using reverse pallet orders is deprecated. use only \
`AllPalletWithSystem or AllPalletsWithoutSystem`")]
```

Reply:
In addition, we should update/remove the comment about this

Reply:
it seems this PR changing it https://github.com/paritytech/substrate/pull/11813. But it does not add much info either..

Reply:
@kianenigma Can you help clarify this change? Just looking at the name change, we went from guaranteeing (1) reversed order and (2) system first, and now the order seems unclear.

In the past, we have found very subtle pallet ordering issues WRT hooks like `on_initialize()` and I want to make sure we aren't going to introduce similar problems.

What we want is the no-op change from `AllPalletsReversedWithSystemFirst`. In our case, we use `pallet_maintenance_mode` in our `pub type Executive ...` declaration, but I'm not sure that matters.

Reply:
For the time being you should still use `AllPalletsWithoutSystemReversed` with `allow(deprecated)`.

Once you have the time, you should consider flipping the order of the pallets in your `construct_runtime` so that you won't need to use the reversed order anymore. 

The best way to test this would be to generate a metadata file before and after making this change, and making sure they are the same. If you maintain pallet order, the metadata should be exactly the same.